### PR TITLE
PR 2 of 40

### PR DESCRIPTION
**Pull request by ahrav**
### Description:
Explain the purpose of the PR.

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/appointedd/appointedd.go [hunk 1]
```diff
@@ -14,66 +14,53 @@ import (
...
-					continue
-				}
-				body := string(bodyBytes)
 
-				if strings.Contains(body, "total") {
-					s1.Verified = true
-				} else {
-					if detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, true) {
-						continue
+				if bodyBytes, err := io.ReadAll(res.Body); err == nil {
```
Comment (dustin-decker) [inline pkg/detectors/appointedd/appointedd.go:51]: Suggestion: https://github.com/trufflesecurity/trufflehog/blob/main/pkg/common/utils.go#L42-L53 would be more efficient

File: pkg/detectors/apptivo/apptivo.go [hunk 2]
```diff
@@ -15,60 +16,52 @@ import (
...
 			if verify {
-				req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("https://api.apptivo.com/app/dao/v6/leads?a=getConfigData&apiKey=%s&accessKey=%s", resMatch, resIdMatch), nil)
+				req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("https://api.apptivo.com/app/dao/v6/leads?a=getConfigData&apiKey=%s&accessKey=%s", string(resMatch), string(resIdMatch)), nil)
 				if err != nil {
 					continue
 				}
 				res, err := client.Do(req)
 				if err == nil {
-					bodyBytes, err := io.ReadAll(res.Body)
+					bodyBytes, err := ioutil.ReadAll(res.Body)
```
Comment (dustin-decker) [inline pkg/detectors/apptivo/apptivo.go:60]: Suggestion: https://github.com/trufflesecurity/trufflehog/blob/main/pkg/common/utils.go#L42-L53 would be more efficient

File: pkg/detectors/atera/atera.go [hunk 3]
```diff
@@ -52,14 +47,14 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
-			req.Header.Add("X-API-KEY", resMatch)
+			req.Header.Add("X-API-KEY", string(resMatch))
 			res, err := client.Do(req)
 			if err == nil {
 				defer res.Body.Close()
 				if res.StatusCode >= 200 && res.StatusCode < 300 {
 					s1.Verified = true
 				} else {
-					// This function will check false positives for common test words, but also it will make sure the key appears 'random' enough to be a real key.
+					// Consider adding the []byte conversion here if the input also changed to a []byte.
```
Comment (dustin-decker) [inline pkg/detectors/atera/atera.go:57]: Nit: comment seems incorrect

File: pkg/detectors/audd/audd.go [hunk 4]
```diff
@@ -26,48 +26,42 @@ var (
...
 
 		if verify {
-			req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("https://api.audd.io/setCallbackUrl/?api_token=%s&url=https://yourwebsite.com/callbacks_handler/", resMatch), nil)
+			req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("https://api.audd.io/setCallbackUrl/?api_token=%s&url=https://yourwebsite.com/callbacks_handler/", string(resMatch)), nil)
 			if err != nil {
 				continue
 			}
 			res, err := client.Do(req)
 			if err == nil {
 				bodyBytes, err := io.ReadAll(res.Body)
```
Comment (dustin-decker) [inline pkg/detectors/audd/audd.go:54]: Suggestion: https://github.com/trufflesecurity/trufflehog/blob/main/pkg/common/utils.go#L42-L53 would be more efficient

File: pkg/detectors/auth0oauth/auth0oauth.go [hunk 5]
```diff
@@ -94,11 +84,7 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 						}
 						body := string(bodyBytes)
 
-						// if client_id and client_secret is valid -> 403 {"error":"invalid_grant","error_description":"Invalid authorization code"}
-						// if invalid -> 401 {"error":"access_denied","error_description":"Unauthorized"}
-						// ingenious!
-
-						if !strings.Contains(body, "access_denied") {
+						if !bytes.Contains(bodyBytes, []byte("access_denied")) {
```
Comment (dustin-decker) [inline pkg/detectors/auth0oauth/auth0oauth.go:87]: Suggestion: https://github.com/trufflesecurity/trufflehog/blob/main/pkg/common/utils.go#L42-L53 would be more efficient

File: pkg/detectors/aws/aws.go [hunk 6]
```diff
@@ -146,17 +145,17 @@ func (s scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
-				params.Add("X-Amz-Credential", resIDMatch+"/"+credentialScope)
+				params.Add("X-Amz-Credential", string(resIDMatch)+"/"+credentialScope)
 				params.Add("X-Amz-Date", amzDate)
 				params.Add("X-Amz-Expires", "30")
 				params.Add("X-Amz-SignedHeaders", signedHeaders)
 
 				canonicalQuerystring := params.Encode()
 				payloadHash := GetHash("") // empty payload
-				canonicalRequest := method + "\n" + canonicalURI + "\n" + canonicalQuerystring + "\n" + canonicalHeaders + "\n" + signedHeaders + "\n" + payloadHash
+				canonicalRequest := string(method) + "\n" + string(canonicalURI) + "\n" + string(canonicalQuerystring) + "\n" + string(canonicalHeaders) + "\n" + string(signedHeaders) + "\n" + string(payloadHash)
```
Comment (dustin-decker) [inline pkg/detectors/aws/aws.go:155]: Many of these are already strings, same on L158

**Review by dustin-decker — CHANGES_REQUESTED** [review]

**Comment by zricethezav** [comment]
Closing this set of PRs until they are all ready to be reviewed and merged

**Pull request closed without merging**